tests/testthat/test-aes.r

```diff
@@ -111,6 +111,41 @@ test_that("aes standardises aesthetic names", {
   expect_warning(aes(color = x, colour = y), "Duplicated aesthetics")
 })
 
+test_that("Improper use of $ and [[ is detected by check_aes_extract_usage()", {
+
+  returns_x <- function() "x"
+  df <- data_frame(x = 1:5, nested_df = data_frame(x = 6:10))
+
+  # valid extraction in aes()
+  expect_silent(check_aes_extract_usage(aes(x), df))
+  expect_silent(check_aes_extract_usage(aes(.data$x), df))
+  expect_silent(check_aes_extract_usage(aes(.data[["x"]]), df))
+  expect_silent(check_aes_extract_usage(aes(.data[[!!quo("x")]]), df))
```

Comment:
I think you can remove the examples with `!!` because `aes()` does the unquoting, so `check_aes_extract_usage()` never sees them.